anyway/infographics_utils.py

```diff
@@ -127,13 +128,16 @@ def get_widget_id(self) -> WidgetId:
 
     # noinspection PyMethodMayBeStatic
     def is_in_cache(self) -> bool:
+        """Whether this widget is stored in the cache"""
         return True
 
     # noinspection PyMethodMayBeStatic
     def is_included(self) -> bool:
+        """Whether this widget is included in the response"""
         return True
 
     def generate_items(self) -> None:
```

Comment:
sorry, just noticed this one, what do you think about making this return type `Optional[object]`?
I don't think we want to alter to widget state so implicitly, so this function will return the generated items, right? In such case the return type *may* be None, but it won't always be.

Reply:
My intention is that generate_items() is a procedure that does alter the state of the Widget instance. It stores its result in the attribute `self.items`. I did this in order to avoid calculation of items multiple times.
Looking at this now, I think we should have a function get_items(), that retrieves the items. It is somewhat redundant, as the attributes of the class are not private anyway.